src/main/java/cz/cvut/kbss/analysis/controller/ComponentController.java

```diff
@@ -59,20 +80,37 @@ public void delete(@PathVariable(name = "componentFragment") String componentFra
         repositoryService.remove(componentUri);
     }
 
+    @Operation(summary = "Retrieve functions of a component")
+    @ApiResponses(value = {
+            @ApiResponse(responseCode = "200", description = "Functions retrieved successfully"),
+            @ApiResponse(responseCode = "404", description = "Component not found")
+    })
     @GetMapping(value = "/{componentFragment}/functions", produces = {JsonLd.MEDIA_TYPE, MediaType.APPLICATION_JSON_VALUE})
     public Set<Function> getFunctions(@PathVariable(name = "componentFragment") String componentFragment) {
         log.info("> getFunctions - {}", componentFragment);
         URI componentUri = identifierService.composeIdentifier(Vocabulary.s_c_component, componentFragment);
         return repositoryService.getFunctions(componentUri);
     }
 
+    @Operation(summary = "Retrieve failure modes of a component")
+    @ApiResponses(value = {
+            @ApiResponse(responseCode = "200", description = "Failure modes retrieved successfully"),
+            @ApiResponse(responseCode = "404", description = "Component not found")
+    })
     @GetMapping(value = "/{componentFragment}/failureModes", produces = {JsonLd.MEDIA_TYPE, MediaType.APPLICATION_JSON_VALUE})
     public Set<FailureMode> getFailureModes(@PathVariable(name = "componentFragment") String componentFragment) {
         log.info("> getFailureModes - {}", componentFragment);
         URI componentUri = identifierService.composeIdentifier(Vocabulary.s_c_component, componentFragment);
         return repositoryService.getFailureModes(componentUri);
     }
 
+
+    @Operation(summary = "Add a failure mode to a component")
+    @ApiResponses(value = {
+            @ApiResponse(responseCode = "201", description = "Failure mode added successfully"),
+            @ApiResponse(responseCode = "400", description = "Invalid input"),
```

Comment:
is it more appropriate to use 409 here?

Reply:
@blcham @palagdan 
For this particular API, there multiple reasons why the request input is bad/invalid. 
- `409(Confilt) - failure mode already exists` - the input failure mode exists in the DB. 
  - This check is not implemented. It should be sufficient to check if the `uri` of the input failure mode exists in the DB. 
- `422(Unprocessable Content) - Missing name` - to create a new failure mode a name property is required.
  - currently the backend will throws `PersistenceException` which is 500

Reply:
@kostobog can you update it in `@ApiResponse(responseCode = "400"` ?

Reply:
@blcham 
I updated the `ApiReposnses`. Added multiple lines for  400 for `failureMode` constraint validations . Also added `ApiResponse` for 404 because which is returned if the `componentFragment` does identify a component.

Reply:
ok thanks.